net/bridge.go

```diff
@@ -338,11 +339,16 @@ func (b bridgeImpl) initPrep(config *BridgeConfig) error {
 	if err := netlink.LinkSetMasterByIndex(dummy, b.bridge.Attrs().Index); err != nil {
 		return errors.Wrap(err, "setting dummy interface master")
 	}
-	if err := netlink.LinkDel(dummy); err != nil {
-		return errors.Wrap(err, "deleting dummy interface")
-	}
-
-	return nil
+	defer func() {
```

Comment:
I'm a bit puzzled here.  The issue talks about using `defer` right after the interface is created.  But here you're two operations later.

Also you're right before `return` so not much value from `defer`.

Am I missing something?

Reply:
So earlier (below snippet), after link creation, failure in either `LinkSetMTU` or `LinkSetMasterByIndex` results in not deleting the interface. When Weave pod restarts, it fails on `LinkAdd` and goes in crashback loop.

```go
	if err = netlink.LinkAdd(dummy); err != nil {
		return errors.Wrap(err, "creating dummy interface")
	}
	if err := netlink.LinkSetMTU(dummy, config.MTU); err != nil {
		return errors.Wrapf(err, "setting dummy interface mtu to %d", config.MTU)
	}
	if err := netlink.LinkSetMasterByIndex(dummy, b.bridge.Attrs().Index); err != nil {
		return errors.Wrap(err, "setting dummy interface master")
	}
	if err := netlink.LinkDel(dummy); err != nil {
		return errors.Wrap(err, "deleting dummy interface")
	}
```

Reply:
Sorry my bad. It should be right after `LinkAdd`, will update.